bmwqemu.pm

```diff
@@ -215,11 +215,17 @@ sub log_call {
     else {
         # key/value pairs
         my @result;
+        my $should_stash = 0;
+        my @tmp_array = @_;
+        while (my ($key, $value) = splice(@tmp_array, 0, 2)) {
```

Comment:
I wouldn't bet on this syntax doing what you want so please provide some unit tests showing it does the right thing. Likely there's also a way to do this without constructing and destructing a temporary array.